Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/StackedNotificationInfo.cs

```diff
@@ -0,0 +1,60 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Base class for information about stacked notification
+    /// </summary>
+    public abstract class StackedNotificationInfo
```

Comment:
Maybe have this have a Generic property of `ContentHolder`?

Reply:
Well. How do you handle a type that is not part of the possible types. In F#, you could do that: `string | UIElement | DataTemplate` but I don't know about generics.

Reply:
```C#
public class StackedNotificationInfo<T>
{
   public int Duration { get; set; }  
   public T ContentHolder { get; set; }
}
```

Then you can do: `public void Show<T>(T contentHolder, int duration = 0)`

Reply:
Is it better than what we have now? I mean, devs could possibly send whatever type they want. We loss the constraints of what I said earlier on restricting to `string | UIElement | DataTemplate` type.

Reply:
Sure, it is possible. But what I am thinking for devs. Is it better to see a method with 3 overloads (with each recommend type) or a single method where you can insert whatever you want?

I know, this is a small argument but I prefer that we look at every angle. If you are still in favor of the generic version, I'll make the change.

Reply:
I don't think we should change the Show methods to take in arbitrary type, they should remain the way they are. I think the discussion here is whether we just need this:

```
internal class StackedNotificationInfo
{
   public int Duration { get; set; }  
   public object Content { get; set; }
}
```

vs what you have right now.

Reply:
I don't think that's what @michael-hawker wanted. But of course, I can change the shape of the internal class.

Reply:
I'm fine with either, the backup behavior @nmetulev suggests makes sense too.